cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesExport.java

```diff
@@ -0,0 +1,408 @@
+        writer.writeCharacters(DateTime.now().toString());
+        writer.writeEndElement();
+        writer.writeStartElement(MD_NAMESPACE, CgmesNames.DESCRIPTION);
+        writer.writeCharacters(context.getSvDescription());
+        writer.writeEndElement();
+        writer.writeStartElement(MD_NAMESPACE, CgmesNames.VERSION);
+        writer.writeCharacters(is(context.getSvVersion()));
+        writer.writeEndElement();
+        for (String dependency : context.getDependencies()) {
+            writer.writeEmptyElement(MD_NAMESPACE, CgmesNames.DEPENDENT_ON);
+            writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, dependency);
+        }
+        writer.writeStartElement(MD_NAMESPACE, CgmesNames.PROFILE);
+        writer.writeCharacters("http://entsoe.eu/CIM/StateVariables/4/1");
+        writer.writeEndElement();
+        writer.writeStartElement(MD_NAMESPACE, CgmesNames.MODELING_AUTHORITY_SET);
+        writer.writeCharacters(context.getModelingAuthoritySet());
+        writer.writeEndElement();
+        writer.writeEndElement();
+    }
+
+    private static void writeTopologicalIslands(Network network, XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
+        if (context.getTopologyKind() == CgmesTopologyKind.NODE_BREAKER) {
+            // TODO we need to export SV file data for NodeBraker
+            LOG.warn("NodeBreaker view require further investigation to map correctly Topological Nodes");
+            return;
+        }
+        Map<String, String> angleRefs = buildAngleRefs(network);
+        Map<String, List<String>> islands = buildIslands(network);
+        for (Map.Entry<String, List<String>> island : islands.entrySet()) {
+            if (!angleRefs.containsKey(island.getKey())) {
+                Supplier<String> log = () -> String.format("Synchronous component  %s does not have a defined slack bus: it is ignored", island.getKey());
+                LOG.info(log.get());
+                continue;
+            }
+            writer.writeStartElement(CIM_NAMESPACE, CgmesNames.TOPOLOGICAL_ISLAND);
+            writer.writeAttribute(RDF_NAMESPACE, ID, getUniqueId());
+            writer.writeStartElement(CIM_NAMESPACE, CgmesNames.NAME);
+            writer.writeCharacters(getUniqueId()); // TODO do we need another name?
+            writer.writeEndElement();
+            writer.writeEmptyElement(CIM_NAMESPACE, "TopologicalIsland.AngleRefTopologicalNode");
+            writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, "#" + angleRefs.get(island.getKey()));
+            for (String tn : island.getValue()) {
+                writer.writeEmptyElement(CIM_NAMESPACE, "TopologicalIsland.TopologicalNodes");
+                writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, "#" + tn);
+            }
+            writer.writeEndElement();
+        }
+    }
+
+    private static Map<String, String> buildAngleRefs(Network network) {
```

Comment:
Maybe put as a comment that it works only for bus branch topology?

Reply:
The implementation is valid also if the IIDM model is node-breaker